src/org/pentaho/di/job/entries/sqoop/AbstractSqoopJobEntry.java

```diff
@@ -315,24 +315,35 @@ protected void executeSqoop( HadoopShim hadoopShim, SqoopShim shim, S config, Co
    * 
    */
   public void configure( HadoopShim shim, S sqoopConfig, Configuration conf ) throws KettleException {
-    try {
-      String dbName, dbUrl, dbUser, dbPassword;
-      if ( sqoopConfig.getModeAsEnum() == JobEntryMode.ADVANCED_LIST ) {
-        dbName = null;
-        dbUrl = sqoopConfig.getConnectFromAdvanced();
-        dbUser = sqoopConfig.getUsernameFromAdvanced();
-        dbPassword = sqoopConfig.getPasswordFromAdvanced();
-      } else {
-        DatabaseMeta databaseMeta = parentJob.getJobMeta().findDatabase( sqoopConfig.getDatabase() );
-        dbName = databaseMeta.getName();
-        dbUrl = databaseMeta.getURL();
-        dbUser = databaseMeta.getUsername();
-        dbPassword = databaseMeta.getPassword();
-      }
-      sqoopConfig.setConnectionInfo( environmentSubstitute( dbName ),
-        environmentSubstitute( dbUrl ), environmentSubstitute( dbUser ),
-        environmentSubstitute( dbPassword ) );
+    configureDatabase( sqoopConfig );
+    configureShim( shim, sqoopConfig, conf );
+  }
+
+  /**
+   * Configure database connection information
+   * @param sqoopConfig - Sqoop configuration
+   */
+  public void configureDatabase( S sqoopConfig ) throws KettleException {
+    DatabaseMeta databaseMeta = getParentJob().getJobMeta().findDatabase( sqoopConfig.getDatabase() );
+
+    // if databaseMeta == null we assume "USE_ADVANCED_MODE" is selected on QUICK_SETUP
```

Comment:
@brosander it's already there - this is why we process only the case when mode is QUICK_SETUP and we were able to identify the named connection.